### PR TITLE
chore(dask-gateway): Install inside of a virtual environment

### PR DESCRIPTION
**Pull request by EyeCantCU**
Installs dask-gateway and dask-gateway-server into a venv

### CODE REVIEW

File: dask-gateway.yaml [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
+  options:
+    # We create a dependency on libpython even though we provide
+    # libpython in the virtual environment. This prevents python
+    # versions on the host from being swapped out. Enabling no-
+    # depends works around this
+    no-depends: true
   dependencies:
     runtime:
-      - python3
+      - python-3
```
Comment (mamccorm) [inline dask-gateway.yaml:16]: [non-blocking] i'd generally align to using 'python3' here, which is more consistent with the other package names. Both are an option and reference the same package

File: dask-gateway.yaml [hunk 2]
```diff
@@ -38,31 +39,67 @@ pipeline:
...
+      ln -s /usr/share/${{package.name}}/.venv/bin/dask-worker ${{targets.destdir}}/usr/bin/dask-worker
 
 subpackages:
   - name: dask-gateway-server
     description: A multi-tenant server for securely deploying and managing Dask clusters
+    options:
+      no-depends: true
+    dependencies:
+      runtime:
+        - python-3
```
Comment (mamccorm) [inline dask-gateway.yaml:74]: [non-blocking] Same as above

File: dask-gateway.yaml [hunk 3]
```diff
@@ -38,31 +39,67 @@ pipeline:
...
+          # Fix venv paths
+          sed -i "s|/home/build|/usr/share|g" ${{targets.subpkgdir}}/usr/share/${{package.name}}-server/.venv/bin/*
+
+          # Include system site-packages
+          sed -i "s|include-system-site-packages = false|include-system-site-packages = true|g" ${{targets.subpkgdir}}/usr/share/${{package.name}}-server/.venv/pyvenv.cfg
+
+          # Symlink scripts to PATH
+          mkdir -p ${{targets.subpkgdir}}/usr/bin/
+          ln -s /usr/share/${{package.name}}-server/.venv/bin/${{package.name}}-jobqueue-launcher ${{targets.subpkgdir}}/usr/bin/${{package.name}}-jobqueue-launcher
+          ln -s /usr/share/${{package.name}}-server/.venv/bin/${{package.name}}-server ${{targets.subpkgdir}}/usr/bin/${{package.name}}-server
```
Comment (mamccorm) [inline dask-gateway.yaml:102]: Could we look at adding some melange tests? especially given we've changed the packaging format

File: dask-gateway.yaml (hunk 1 above)
Comment (EyeCantCU) [reply dask-gateway.yaml:16]: Thank you. I'll get that switched up :)

File: dask-gateway.yaml (hunk 3 above)
Comment (EyeCantCU) [reply dask-gateway.yaml:102]: Yeah. I can do that

Comment (EyeCantCU) [reply dask-gateway.yaml:102]: Added :)

**Review by mamccorm — APPROVED** [review]

**Pull request merged**